src/main/java/org/broadinstitute/hellbender/utils/mcmc/PosteriorSummary.java

```diff
@@ -1,28 +1,117 @@
+                                                                           final JavaSparkContext ctx) {
+        Utils.nonNull(samples);
+        Utils.validateArg(samples.size() > 0, "Number of samples must be greater than zero.");
+        Utils.validateArg(0 < alpha && alpha < 1, "Alpha must be in (0, 1).");
+
+        final double central = calculatePosteriorMode(samples, ctx);
+
+        //find highest-posterior-density interval using simple Chen & Shao 1998 procedure
+        final List<Double> sortedSamples = new ArrayList<>(samples);
+        Collections.sort(sortedSamples);
+        final int n = sortedSamples.size();
+        double lower = sortedSamples.get(0);
+        double upper = sortedSamples.get(n - 1);
+        double minIntervalWidth = sortedSamples.get(n - 1) - sortedSamples.get(0);
+        final int numSamplesInInterval = (int) Math.floor((1 - alpha) * n);
+        for (int i = 0; i < n - numSamplesInInterval; i++) {
+            final double intervalWidth = sortedSamples.get(i + numSamplesInInterval) - sortedSamples.get(i);
+            if (intervalWidth < minIntervalWidth) {
+                minIntervalWidth = intervalWidth;
+                lower = sortedSamples.get(i);
+                upper = sortedSamples.get(i + numSamplesInInterval);
+            }
+        }
+
+        return new PosteriorSummary(central, lower, upper);
     }
 
-    public double standardDeviation() {
-        return standardDeviation;
+    /**
+     * Given a list of posterior samples, returns an estimate of the posterior mode (using
+     * mllib kernel density estimation in {@link KernelDensity} and {@link BrentOptimizer}).
+     * Note that estimate may be poor if number of samples is small (resulting in poor kernel density estimation),
+     * or if posterior is not unimodal (or is sufficiently pathological otherwise).
+     * @param samples   posterior samples, cannot be {@code null} and number of samples must be greater than 0
+     * @param ctx       {@link JavaSparkContext} used by {@link KernelDensity} for mllib kernel density estimation
+     */
+    public static double calculatePosteriorMode(final List<Double> samples, final JavaSparkContext ctx) {
+        Utils.nonNull(samples);
+        Utils.validateArg(samples.size() > 0, "Number of samples must be greater than zero.");
+        final double sampleMin = Collections.min(samples);
+        final double sampleMax = Collections.max(samples);
+        final double sampleMean = new Mean().evaluate(Doubles.toArray(samples));
+        final double sampleStandardDeviation = new StandardDeviation().evaluate(Doubles.toArray(samples));
+        if (sampleStandardDeviation == 0.0 || Double.isNaN(sampleMean)) {
+            return sampleMean;
+        }
+        //use Silverman's rule to set bandwidth for kernel density estimation from sample standard deviation
+        final double bandwidth = 1.06 * sampleStandardDeviation * Math.pow(samples.size(), -0.2);
```

Comment:
i wouldn't expect users to look for an extracted method in this class if they were to need it elsewhere (say for histogram binning, etc.), so i guess i will go with extracting static constants.  i'm not sure if the case is that strong for extracting here (since i think magic number-ness is sufficiently demystified by the comment and i don't think these numbers will need to be used elsewhere or changed, e.g., to more sig figs, in this class in the future), but better safe than sorry, i guess!  i also added some more comments throughout this method.